trackpy/feature.py

```diff
@@ -571,27 +582,51 @@ def batch(frames, diameter, minmass=100, maxsize=None, separation=None,
             # Interpret meta to be a file handle.
             record_meta(meta_info, meta)
 
-    all_features = []
-    for i, image in enumerate(frames):
-        features = locate(image, diameter, minmass, maxsize, separation,
-                          noise_size, smoothing_size, threshold, invert,
-                          percentile, topn, preprocess, max_iterations,
-                          filter_before, filter_after, characterize,
-                          engine)
-        if hasattr(image, 'frame_no') and image.frame_no is not None:
-            frame_no = image.frame_no
-            # If this works, locate created a 'frame' column.
-        else:
-            frame_no = i
-            features['frame'] = i  # just counting iterations
-        logger.info("Frame %d: %d features", frame_no, len(features))
-        if len(features) == 0:
-            continue
+    # Prepare wrapped function for mapping to `frames`
+    curried_locate = partial(
+        locate,
+        diameter=diameter, minmass=minmass, maxsize=maxsize,
+        separation=separation, noise_size=noise_size,
+        smoothing_size=smoothing_size, threshold=threshold, invert=invert,
+        percentile=percentile, topn=topn, preprocess=preprocess,
+        max_iterations=max_iterations, filter_before=filter_before,
+        filter_after=filter_after, characterize=characterize, engine=engine
+    )
+    # Choose threaded or normal map function
+    if threads == 0:
+        pool = None
+        map_func = map
+    else:
+        pool = ThreadPool(processes=threads)
```

Comment:
Is there a reasoning behind why you chose `ThreadPool`? I switched it to `Pool` and got much better performance (7 sec for `Pool` instead of 14 sec for `ThreadPool`, 16 sec with no threading at all)

Reply:
No, there is no specific reason. It was just an approach that worked for another use case in another project and I implicitly thought that threading would be enough to take advantage of NumPy's and Numba's GIL-releasing features while still being lightweight. 

My understanding is that multiprocessing increases the memory footprint and has a larger overhead compared to threading? But a speedup of 2 is certainly worth it in my opinion.

Reply:
A factor 2 is absolutely worth it. Memory consumption is not an issue in this case.

Reply:
I tested this myself and there was no noticeable difference between the threading and multiprocessing version. So I think this depends on the setup and processed data itself.  For now I went with multiprocessing.